[GTK][WPE] Get rid of non-DisplayLink code path

nikolaszimmermann opened the pull request:
#### d46abace846948d54a798fb965ccefeea1e54dbd
<pre>
[GTK][WPE] Get rid of non-DisplayLink code path
<a href="https://bugs.webkit.org/show_bug.cgi?id=300044">https://bugs.webkit.org/show_bug.cgi?id=300044</a>

Reviewed by Fujii Hironori.

The &quot;#if !HAVE(DISPLAY_LINK)&quot; code path is no longer used for Gtk/WPE ports.
Only the PlayStation port still uses it by default -- therefore split
LayerTreeHost/ThreadedCompositor into parts that use only the
&quot;#if !HAVE(DISPLAY_LINK)&quot; code path (with ifdefs removed) in
LayeTreeHostPlayStation/ThreadedCompositorPlayStation.

This allows us to remove the code path entirely from our LayerTreeHost
and ThreadedCompositor. We can also remove all PLATFORM(WPE) || PLATFORM(GTK)
checks since those classes are now exclusively used by Gtk/WPE ports,
simplifying the code.

Covered by existing tests.

* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CoordinatedSceneState.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::enterAcceleratedCompositingMode):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::layerFlushTimerFired):
(WebKit::LayerTreeHost::forceRepaint):
(WebKit::LayerTreeHost::forceRepaintAsync):
(WebKit::LayerTreeHost::ensureDrawing):
(WebKit::LayerTreeHost::didComposite):
(WebKit::LayerTreeHost::createDisplayRefreshMonitor): Deleted.
(WebKit::LayerTreeHost::requestDisplayRefreshMonitorUpdate): Deleted.
(WebKit::LayerTreeHost::handleDisplayRefreshMonitorUpdate): Deleted.
(WebKit::LayerTreeHost::renderNextFrame): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostPlayStation.cpp: Copied from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp.
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::~LayerTreeHost):
(WebKit::LayerTreeHost::setLayerTreeStateIsFrozen):
(WebKit::LayerTreeHost::scheduleLayerFlush):
(WebKit::LayerTreeHost::cancelPendingLayerFlush):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::layerFlushTimerFired):
(WebKit::LayerTreeHost::updateRootLayer):
(WebKit::LayerTreeHost::setRootCompositingLayer):
(WebKit::LayerTreeHost::setViewOverlayRootLayer):
(WebKit::LayerTreeHost::forceRepaint):
(WebKit::LayerTreeHost::forceRepaintAsync):
(WebKit::LayerTreeHost::sizeDidChange):
(WebKit::LayerTreeHost::pauseRendering):
(WebKit::LayerTreeHost::resumeRendering):
(WebKit::LayerTreeHost::graphicsLayerFactory):
(WebKit::LayerTreeHost::visibleContentsRect const):
(WebKit::LayerTreeHost::backgroundColorDidChange):
(WebKit::LayerTreeHost::attachLayer):
(WebKit::LayerTreeHost::detachLayer):
(WebKit::LayerTreeHost::notifyCompositionRequired):
(WebKit::LayerTreeHost::isCompositionRequiredOrOngoing const):
(WebKit::LayerTreeHost::requestComposition):
(WebKit::LayerTreeHost::compositingRunLoop const):
(WebKit::LayerTreeHost::paintingEngine):
(WebKit::LayerTreeHost::imageBackingStore):
(WebKit::LayerTreeHost::createGraphicsLayer):
(WebKit::LayerTreeHost::createDisplayRefreshMonitor):
(WebKit::LayerTreeHost::requestDisplayRefreshMonitorUpdate):
(WebKit::LayerTreeHost::handleDisplayRefreshMonitorUpdate):
(WebKit::LayerTreeHost::willRenderFrame):
(WebKit::LayerTreeHost::didRenderFrame):
(WebKit::LayerTreeHost::didComposite):
(WebKit::LayerTreeHost::commitSceneState):
(WebKit::LayerTreeHost::renderNextFrame):
(WebKit::LayerTreeHost::notifyFrameDamageForTesting):
(WebKit::LayerTreeHost::resetDamageHistoryForTesting):
(WebKit::LayerTreeHost::foreachRegionInDamageHistoryForTesting):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostPlayStation.h: Copied from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::m_compositingRunLoop):
(WebKit::m_didRenderFrameTimer):
(WebKit::ThreadedCompositor::invalidate):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::frameComplete):
(WebKit::ThreadedCompositor::didRenderFrameTimerFired):
(WebKit::m_displayRefreshMonitor): Deleted.
(WebKit::ThreadedCompositor::displayRefreshMonitor const): Deleted.
(WebKit::ThreadedCompositor::displayUpdateFired): Deleted.
(WebKit::ThreadedCompositor::sceneUpdateFinished): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.cpp: Copied from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp.
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::m_compositingRunLoop):
(WebKit::m_displayRefreshMonitor):
(WebKit::ThreadedCompositor::surfaceID const):
(WebKit::ThreadedCompositor::invalidate):
(WebKit::ThreadedCompositor::suspend):
(WebKit::ThreadedCompositor::resume):
(WebKit::ThreadedCompositor::isActive const):
(WebKit::ThreadedCompositor::backgroundColorDidChange):
(WebKit::ThreadedCompositor::setSize):
(WebKit::ThreadedCompositor::setDamagePropagationFlags):
(WebKit::ThreadedCompositor::enableFrameDamageNotificationForTesting):
(WebKit::ThreadedCompositor::updateSceneState):
(WebKit::ThreadedCompositor::paintToCurrentGLContext):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::requestComposition):
(WebKit::ThreadedCompositor::scheduleUpdate):
(WebKit::ThreadedCompositor::runLoop):
(WebKit::ThreadedCompositor::frameComplete):
(WebKit::ThreadedCompositor::didRenderFrameTimerFired):
(WebKit::ThreadedCompositor::displayRefreshMonitor const):
(WebKit::ThreadedCompositor::displayUpdateFired):
(WebKit::ThreadedCompositor::sceneUpdateFinished):
(WebKit::ThreadedCompositor::updateSceneAttributes):
(WebKit::ThreadedCompositor::initializeFPSCounter):
(WebKit::ThreadedCompositor::updateFPSCounter):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.h: Copied from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.h.
(WebKit::ThreadedCompositor::fps const):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedDisplayRefreshMonitorPlayStation.cpp: Renamed from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedDisplayRefreshMonitor.cpp.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedDisplayRefreshMonitorPlayStation.h: Renamed from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedDisplayRefreshMonitor.h.

Canonical link: <a href="https://commits.webkit.org/301052@main">https://commits.webkit.org/301052@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d1c33485b13bc739b0be5b5572f4ef896a143bb2

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/124807 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/44479 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/35214 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/131652 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/76716 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45182 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/53048 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/94962 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/63003 "") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/127761 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/36040 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/111617 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/75530 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/34972 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/75128 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/105797 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/30001 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/134319 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/51652 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/39455 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/103439 "12 flakes 55 failures") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/52067 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/107836 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/103209 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26278 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/48574 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/26858 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/48657 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/51526 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/57326 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/50919 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/54272 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/52613 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b1d4e23c10de8d3b2fb40a384d3aeca8a90c18c5. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51697

carlosgcampos commented on Tools/Scripts/webkitpy/style/checker.py:
```diff
@@ -343,6 +343,10 @@
      os.path.join('Source', 'WebKit', 'Shared', 'Extensions', 'WebExtensionSQLiteHelpers.h')],
      ["-readability/naming/underscores"]),
 
+    ([  # This file triggers false positives in the constructor initialization list (due to lambda usage), which looks like wrongly indented namespace code to the style checker
+     os.path.join('Source', 'WebKit', 'WebProcess', 'WebPage', 'CoordinatedGraphics', 'ThreadedCompositorPlayStation.cpp')],
+     ["-whitespace/indent"]),
```
This is a common false positive, I would just ignore the style failure instead of ignoring all possible indent issues in that file.

carlosgcampos commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.cpp:
```diff
@@ -0,0 +1,465 @@
+#include <wtf/SystemTracing.h>
+#include <wtf/TZoneMallocInlines.h>
+
+#if USE(GLIB_EVENT_LOOP)
+#include <wtf/glib/RunLoopSourcePriority.h>
+#endif
```
This can be removed

carlosgcampos commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.cpp:
```diff
@@ -0,0 +1,465 @@
+namespace WebKit {
+using namespace WebCore;
+
+static constexpr unsigned c_defaultRefreshRate = 60000;
```
We need to keep the display link ifdefs in the ps implementation, since they use display link optionally when building with `USE_WPE_BACKEND_PLAYSTATION`

carlosgcampos commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.cpp:
```diff
@@ -0,0 +1,465 @@
+
+    m_compositingRunLoop->performTaskSync([this, protectedThis = Ref { *this }] {
+        m_display.updateTimer = makeUnique<RunLoop::Timer>(RunLoop::currentSingleton(), "ThreadedCompositor::UpdateTimer"_s, this, &ThreadedCompositor::displayUpdateFired);
+#if USE(GLIB_EVENT_LOOP)
+        m_display.updateTimer->setPriority(RunLoopSourcePriority::CompositingThreadUpdateTimer);
+#endif
```
This can be removed too.

carlosgcampos commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.cpp:
```diff
@@ -0,0 +1,465 @@
+{
+    ASSERT(m_sceneState);
+    ASSERT(m_compositingRunLoop->isCurrent());
+#if PLATFORM(GTK) || PLATFORM(WPE)
+    TraceScope traceScope(RenderLayerTreeStart, RenderLayerTreeEnd);
+#endif
```
And this

carlosgcampos commented on Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.cpp:
```diff
@@ -0,0 +1,465 @@
+void ThreadedCompositor::updateFPSCounter()
+{
+    if (!m_fpsCounter.exposesFPS
+#if USE(SYSPROF_CAPTURE)
+        && !SysprofAnnotator::singletonIfCreated()
+#endif
```
I guess ps doesn't support sysprof either

carlosgcampos reviewed (commented):
I wonder if ps port could switch to use WPE_BACKEND_PLAYSTATION unconditionally

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/43a395b52aca70c398387316a7718c63c2880ee7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51697

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d1c33485b13bc739b0be5b5572f4ef896a143bb2. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51697

nikolaszimmermann committed:
[GTK][WPE] Get rid of non-DisplayLink code path
https://bugs.webkit.org/show_bug.cgi?id=300044

Reviewed by Fujii Hironori.

The "#if !HAVE(DISPLAY_LINK)" code path is no longer used for Gtk/WPE ports.
Only the PlayStation port still uses it by default -- therefore split
LayerTreeHost/ThreadedCompositor into parts that use only the
"#if !HAVE(DISPLAY_LINK)" code path (with ifdefs removed) in
LayeTreeHostPlayStation/ThreadedCompositorPlayStation.

This allows us to remove the code path entirely from our LayerTreeHost
and ThreadedCompositor. We can also remove all PLATFORM(WPE) || PLATFORM(GTK)
checks since those classes are now exclusively used by Gtk/WPE ports,
simplifying the code.

Covered by existing tests.

* Source/WebKit/PlatformPlayStation.cmake:
* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/CoordinatedSceneState.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/DrawingAreaCoordinatedGraphics.cpp:
(WebKit::DrawingAreaCoordinatedGraphics::enterAcceleratedCompositingMode):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp:
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::layerFlushTimerFired):
(WebKit::LayerTreeHost::forceRepaint):
(WebKit::LayerTreeHost::forceRepaintAsync):
(WebKit::LayerTreeHost::ensureDrawing):
(WebKit::LayerTreeHost::didComposite):
(WebKit::LayerTreeHost::createDisplayRefreshMonitor): Deleted.
(WebKit::LayerTreeHost::requestDisplayRefreshMonitorUpdate): Deleted.
(WebKit::LayerTreeHost::handleDisplayRefreshMonitorUpdate): Deleted.
(WebKit::LayerTreeHost::renderNextFrame): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostPlayStation.cpp: Copied from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.cpp.
(WebKit::LayerTreeHost::LayerTreeHost):
(WebKit::LayerTreeHost::~LayerTreeHost):
(WebKit::LayerTreeHost::setLayerTreeStateIsFrozen):
(WebKit::LayerTreeHost::scheduleLayerFlush):
(WebKit::LayerTreeHost::cancelPendingLayerFlush):
(WebKit::LayerTreeHost::flushLayers):
(WebKit::LayerTreeHost::layerFlushTimerFired):
(WebKit::LayerTreeHost::updateRootLayer):
(WebKit::LayerTreeHost::setRootCompositingLayer):
(WebKit::LayerTreeHost::setViewOverlayRootLayer):
(WebKit::LayerTreeHost::forceRepaint):
(WebKit::LayerTreeHost::forceRepaintAsync):
(WebKit::LayerTreeHost::sizeDidChange):
(WebKit::LayerTreeHost::pauseRendering):
(WebKit::LayerTreeHost::resumeRendering):
(WebKit::LayerTreeHost::graphicsLayerFactory):
(WebKit::LayerTreeHost::visibleContentsRect const):
(WebKit::LayerTreeHost::backgroundColorDidChange):
(WebKit::LayerTreeHost::attachLayer):
(WebKit::LayerTreeHost::detachLayer):
(WebKit::LayerTreeHost::notifyCompositionRequired):
(WebKit::LayerTreeHost::isCompositionRequiredOrOngoing const):
(WebKit::LayerTreeHost::requestComposition):
(WebKit::LayerTreeHost::compositingRunLoop const):
(WebKit::LayerTreeHost::paintingEngine):
(WebKit::LayerTreeHost::imageBackingStore):
(WebKit::LayerTreeHost::createGraphicsLayer):
(WebKit::LayerTreeHost::createDisplayRefreshMonitor):
(WebKit::LayerTreeHost::requestDisplayRefreshMonitorUpdate):
(WebKit::LayerTreeHost::handleDisplayRefreshMonitorUpdate):
(WebKit::LayerTreeHost::willRenderFrame):
(WebKit::LayerTreeHost::didRenderFrame):
(WebKit::LayerTreeHost::didComposite):
(WebKit::LayerTreeHost::commitSceneState):
(WebKit::LayerTreeHost::renderNextFrame):
(WebKit::LayerTreeHost::notifyFrameDamageForTesting):
(WebKit::LayerTreeHost::resetDamageHistoryForTesting):
(WebKit::LayerTreeHost::foreachRegionInDamageHistoryForTesting):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostPlayStation.h: Copied from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHost.h.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp:
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::m_compositingRunLoop):
(WebKit::m_didRenderFrameTimer):
(WebKit::ThreadedCompositor::invalidate):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::frameComplete):
(WebKit::ThreadedCompositor::didRenderFrameTimerFired):
(WebKit::m_displayRefreshMonitor): Deleted.
(WebKit::ThreadedCompositor::displayRefreshMonitor const): Deleted.
(WebKit::ThreadedCompositor::displayUpdateFired): Deleted.
(WebKit::ThreadedCompositor::sceneUpdateFinished): Deleted.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.h:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.cpp: Copied from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp.
(WebKit::ThreadedCompositor::create):
(WebKit::ThreadedCompositor::ThreadedCompositor):
(WebKit::m_compositingRunLoop):
(WebKit::m_displayRefreshMonitor):
(WebKit::ThreadedCompositor::surfaceID const):
(WebKit::ThreadedCompositor::invalidate):
(WebKit::ThreadedCompositor::suspend):
(WebKit::ThreadedCompositor::resume):
(WebKit::ThreadedCompositor::isActive const):
(WebKit::ThreadedCompositor::backgroundColorDidChange):
(WebKit::ThreadedCompositor::setSize):
(WebKit::ThreadedCompositor::setDamagePropagationFlags):
(WebKit::ThreadedCompositor::enableFrameDamageNotificationForTesting):
(WebKit::ThreadedCompositor::updateSceneState):
(WebKit::ThreadedCompositor::paintToCurrentGLContext):
(WebKit::ThreadedCompositor::renderLayerTree):
(WebKit::ThreadedCompositor::requestComposition):
(WebKit::ThreadedCompositor::scheduleUpdate):
(WebKit::ThreadedCompositor::runLoop):
(WebKit::ThreadedCompositor::frameComplete):
(WebKit::ThreadedCompositor::didRenderFrameTimerFired):
(WebKit::ThreadedCompositor::displayRefreshMonitor const):
(WebKit::ThreadedCompositor::displayUpdateFired):
(WebKit::ThreadedCompositor::sceneUpdateFinished):
(WebKit::ThreadedCompositor::updateSceneAttributes):
(WebKit::ThreadedCompositor::initializeFPSCounter):
(WebKit::ThreadedCompositor::updateFPSCounter):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositorPlayStation.h: Copied from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.h.
(WebKit::ThreadedCompositor::fps const):
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedDisplayRefreshMonitorPlayStation.cpp: Renamed from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedDisplayRefreshMonitor.cpp.
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedDisplayRefreshMonitorPlayStation.h: Renamed from Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedDisplayRefreshMonitor.h.

Canonical link: https://commits.webkit.org/301052@main
